### ISSUE TITLE
Install bit32 library on Lua 5.3

### ISSUE THREAD

**Issue by moteus** [keplerproject/luarocks#526]
LuaRocks does not allow install `bit32` library on Lua 5.3.
```
>luarocks-5.3 install bit32

Error: Rock bit32 5.3-1 was found but it is provided by VM or 'rocks_provided' in the config file.
```
Yes Lua 5.3 has bit operatinos. But some modules still uses bit library and does not update to use bit operations.

**Comment by siffiejoe** [comment]
The bit32 module is built-in into Lua 5.3 unless you disable the compatibility flags.

**Comment by moteus** [comment]
I know that. And suppose I have Lua without this library. For me easy way is install `bit32` library instead of rebuild Lua.

**Comment by tst2005** [comment]
Hello,
This [cheap bit32 emulation](http://lua-users.org/lists/lua-l/2015-05/msg00120.html) may help you.

**Comment by moteus** [comment]
Problem not how implement it. I know how to do it and I do it for my modules. But if i use not my module which uses bit32 library. This is not my module so I can not edit it on LuaRocks and author does not release this changes. So I can not use this module because LuaRocks does not allow me install bit32 library.

**Comment by mpeterv** [comment]
Basically LuaRocks shouldn't consider `bit32` "provided" under Lua 5.3 with no compatibility flags.

**Comment by hishamhm** [comment]
so, this?

```diff
+++ b/src/luarocks/cfg.lua
@@ -603,13 +603,11 @@ defaults.variables.LUA = site_config.LUA_DIR_SET and (defa
ults.variables.LUA_BIN
 -- Add built-in modules to rocks_provided
 defaults.rocks_provided["lua"] = cfg.lua_version.."-1"
 
-if cfg.lua_version >= "5.2" then
-   -- Lua 5.2+
+if bit32 then -- Lua 5.2 (and Lua 5.3 with compat flags)
    defaults.rocks_provided["bit32"] = cfg.lua_version.."-1"
 end
 
-if cfg.lua_version >= "5.3" then
-   -- Lua 5.3+
+if utf8 then -- Lua 5.3
    defaults.rocks_provided["utf8"] = cfg.lua_version.."-1"
 end
 
```

**Comment by mpeterv** [comment]
Fixed in @93f594a

**Comment by hishamhm** [comment]
Thanks for catching this, @mpeterv!
